File: src/main/java/net/scriptgate/masonry/api/Brick.java
package net.scriptgate.masonry.api;

public interface Brick {

    void goTo(int x, int y);

    default void moveTo(int x, int y) {
        goTo(x, y);
    }

    int getHeight();

    int getWidth();

    boolean isLayoutInstant();

    int getX();

    int getY();
}

File: src/main/kotlin/net/scriptgate/masonry/api/Brick.kt
package net.scriptgate.masonry.api

interface Brick {

    val height: Int

    val width: Int

    val isLayoutInstant: Boolean

    val x: Int

    val y: Int

    fun goTo(x: Int, y: Int)

    open fun moveTo(x: Int, y: Int) {
        goTo(x, y)
    }
}
